mti_nma/steps/nma/nma.py

```diff
@@ -90,6 +89,9 @@ def run(
             Default: None (no distributed executor, use local threads)
         """
 
+        # Set upstream task using structure-named version of previous step
+        self.direct_upstream_tasks = [Avgshape(step_name=f"avgshape_{struct}")]
```

Comment:
I don't think this will work...

Reply:
Hm - then how am I supposed to use the structure-named (non-generic) upstream tasks so that pull will work?

Reply:
Right, this is why I think `pull` is currently broken. This is set in `run`. Not `init`. The way step works is that when you want to do any operation. You init the class then run it there is no statefulness between commands.

Reply:
Okay I understand - thanks